src/status_im2/contexts/share/view.cljs

```diff
@@ -0,0 +1,168 @@
+(ns status-im2.contexts.share.view
+  (:require [utils.i18n :as i18n]
+            [quo2.core :as quo]
+            [react-native.core :as rn]
+            [react-native.safe-area :as safe-area]
+            [status-im2.contexts.share.style :as style]
+            [utils.re-frame :as rf]
+            [reagent.core :as reagent]
+            [quo2.foundations.colors :as colors]
+            [quo2.components.qr-code.qr :as qr]
+            [clojure.string :as string]
+            ;;TODO(siddarthkay) : move the components below over to status-im2 ns
+            ;; issue -> https://github.com/status-im/status-mobile/issues/15549
+            [status-im.ui.components.list-selection :as list-selection]
+            [status-im.ui.components.react :as react]))
+
+(def ^:const profile-tab-id 0)
+(def ^:const wallet-tab-id 1)
+
+(defn copy-text-and-show-toast
+  [{:keys [text-to-copy post-copy-message]}]
+  (react/copy-to-clipboard text-to-copy)
+  (js/setTimeout
+   #(rf/dispatch [:share/show-successfully-copied-toast post-copy-message])
+   250))
+
+(defn header
+  []
+  [rn/view
+   [quo/button
+    {:icon                true
+     :type                :blur-bg
+     :size                32
+     :accessibility-label :close-activity-center
+     :override-theme      :dark
+     :style               style/header-button
+     :on-press            #(rf/dispatch [:hide-popover])}
+    :i/close]
+   [quo/text
+    {:size   :heading-1
+     :weight :semi-bold
+     :style  style/header-heading}
+    (i18n/label :t/share)]])
+
+(defn profile-tab
+  [window-width]
+  (let [multiaccount              (rf/sub [:multiaccount])
+        emoji-hash                (string/join "" (get multiaccount :emoji-hash))
+        qr-size                   (- window-width 64)
+        public-pk                 (get multiaccount :public-key)
+        profile-qr-url            (str "https://join.status.im/u/" public-pk)
+        port                      (rf/sub [:mediaserver/port])
+        key-uid                   (get multiaccount :key-uid)
+        emoji-hash-max-width      (* window-width 0.76)
```

Comment:
window-width should be never used, only margins and paddings should be used

Reply:
the link to a users profile looks like this :
`https://join.status.im/u/0x042286528b6970606f57c3366d7157e3ce7ed18d3f333ce594cfd4d804a16343c7153f114a553b6d9a73bc48ddcf79e31d959dfaa5dc3139711f862c237552ffb0` 

see : https://github.com/status-im/status-mobile/pull/15528/files#diff-bae9803f75568672a9503a5d292a4568f94718d737e349dcd02826d822228648R84
and : https://github.com/status-im/status-mobile/pull/15528/files#diff-dc8d3223adba2ef598af84ebc78534df38654d4b5cb89ea6dd0990b6c40a56aeR59

We use `:ellipsize-mode  :middle` to shorten it. 
This will only work if we know exactly the space we have for the complete sentence. 
Now I could use a fixed and definite number like 263 from figma OR rely on a percentage of window-width.

I think in this area using window-width is fine because what this means is that fit the url into 76% of screen width and there is nothing we can do with margin or padding.

Is this okay after the explanation on why I chose this approach? if not can you explain another better way to do this?

Reply:
> This will only work if we know exactly the space we have for the complete sentence.

this sounds strange

Reply:
for context sake we are talking about `max-width` of this component 
<img width="1048" alt="Screenshot 2023-04-04 at 12 46 04 PM" src="https://user-images.githubusercontent.com/64726664/229716828-9d8f3c68-6064-4aed-9363-459471956027.png">

also to cc : @OmarBasem  @smohamedjavid

Reply:
I would agree with @flexsurfer here, you should be able to achieve everything here with padding and flex box. Perhaps some heights have to be set but I don't think you need to use window-width.

Also I noticed that we have basically this component in quo2 library. - https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?node-id=1338-34507&t=dgFUSXeZh0hk0GFV-0

Maybe we can make a pr after this is merged in and abstract this component to quo2?

Reply:
@J-Son89 : To add more context I have highlighted the respective backgrounds with success and danger colours.

![Screenshot 2023-04-04 at 3 16 28 PM](https://user-images.githubusercontent.com/64726664/229754124-39dfa796-c330-4efc-a669-0bf6cc3bf02a.png)

Now we want to restrict the section in red so that it does not push the share button on the right side out of the desired place.
The text inside has the following value `https://join.status.im/u/0x042286528b6970606f57c3366d7157e3ce7ed18d3f333ce594cfd4d804a16343c7153f114a553b6d9a73bc48ddcf79e31d959dfaa5dc3139711f862c237552ffb0` which is 158 characters 🫣

And we want to show an ellipsis, the only way we can restrict that is either by giving a fixed width to the text component or to its container view. 
Now I chose max-width to do that, can we achieve the same thing with flex box and padding?

Reply:
Hi @siddarthkay,
Something like the following would achieve that design:

```
[rn/view {:style {:margin-horizontal 20
                           :flex-direction  :row
                           :align-items :center}}
   [rn/text {:ellipsize-mode :middle
                  :number-of-lines 1
                   :style {:color :black
                              :margin-right 10}} "some long text"]
   [button]]
```

Reply:
I don't have anything to add, but I agree with @J-Son89 and @flexsurfer.

From the docs:

>middle - The line is displayed so that the beginning and end fit in the container and the missing text in the middle is indicated by an ellipsis glyph. "ab...yz"

So as others have pointed out, the link should adjust to the parent container if margins and paddings are used, but maybe I'm missing something.

Reply:
@OmarBasem : your approach works well when there is one container but if you look at the code below we have 2 container views which have flex-direction as column :) that complicates things a bit when you have a really long text and when you do not set its width explicitly.

We want those columns to have the ability to centre align the button vertically..
 
 ref code  : 
https://github.com/status-im/status-mobile/blob/fc677da58a12c457739ac26d58910ee438e7b9c8/src/status_im2/contexts/share/view.cljs#L51-L88
         
ref styles : 
https://github.com/status-im/status-mobile/blob/fc677da58a12c457739ac26d58910ee438e7b9c8/src/status_im2/contexts/share/style.cljs#L30-L83

Reply:
One compromise I can agree with right now is to give the container a fixed width of 263 as mentioned in figma : https://www.figma.com/file/eDfxTa9IoaCMUy5cLTp0ys/Shell-for-Mobile?node-id=451-18993&t=aMXl84GPyzbQ8A6W-0 
This will ensure that the content does not overflow

Reply:
@siddarthkay I don't see the problem with that. You can have as many containers as you want:

```
[rn/view {:style {:margin-horizontal 20
                    :flex-direction  :row
                    :justify-content :space-between
                    :align-items :center}}
  [rn/view {:style {:flex-direction :column}}
   [rn/text "TITLE 1"]
   [rn/text {:ellipsize-mode :middle
             :number-of-lines 1
             :style {:color :black
                     :margin-right 10}} "some long aksdflkasdkljflkasdfl;kjasl;kdjfl;ajsdflkjasldkjflkjasdfkljjk;ltext"]]
   [rn/view {:style {:flex-direction :column}}
    [audio-button]]]
```

Reply:
fixed here : https://github.com/status-im/status-mobile/pull/15528/commits/9954eb81059ec4ddf4eb8d7317c6de820531dcbc

Thanks everyone for indulging 😁